packages/dd-trace/src/util.js

```diff
@@ -87,12 +87,83 @@ function generatePointerHash (components) {
   return hash.substring(0, 32)
 }
 
+/**
+ * Encodes a DynamoDB attribute value to Buffer for span pointer hashing.
+ * @param {Object} valueObject - DynamoDB value in AWS format ({ S: string } or { N: string } or { B: Buffer })
```

Comment:
Yes. In SDK V3, the user is required to marshal their item before submitting the PutItem request.

In SDK V2, the user does not have to marshal the item, but if they don't, it will automatically be marshaled into this expected format before we receive it.